pkg/app/registry/static/static.go

```diff
@@ -22,39 +22,84 @@ import (
 	"context"
 	"strings"
 
+	registrypb "github.com/cs3org/go-cs3apis/cs3/app/registry/v1beta1"
 	"github.com/cs3org/reva/pkg/app"
+	"github.com/cs3org/reva/pkg/app/registry/registry"
 	"github.com/cs3org/reva/pkg/errtypes"
 	"github.com/cs3org/reva/pkg/sharedconf"
 	"github.com/mitchellh/mapstructure"
 )
 
-type registry struct {
-	rules map[string]string
+func init() {
+	registry.Register("static", New)
+}
+
+type config struct {
+	Providers map[string]*registrypb.ProviderInfo `mapstructure:"providers"`
```

Comment:
@ishank011 I don't fully get the strategy how to get a list of supported mime types by client applications (not by the app provider itself). With the current implementation I would need to register an app provider multiple times (depending on the count of client applications) and change the description of each instance to the client application's name.

Example: 
The app provider "wopi" supports opening files in different client applications: "Collabora", "Microsoft Office Online Server", "CodiMD" and "Etherpad". As a user I don't want to see that I can open the file with "wopi" but I want to see if I can open the file in "Collabora", ... And I only want to see the file action if the open action is possible for that client application.

Reply:
Hi @wkloucek. Yes you would need multiple instances of the appprovider, each pointing to a different client app, all of which can be served by the same wopi server though. At initialization, each of the appproviders will call their discovery URLs, retrieve the mime types and register themselves as  "Collabora", "Microsoft Office Online Server", "CodiMD", and so on. From the web app, we can call `FindAppProviders` with the file name, which will give you a list of all apps supporting that mimetype. Then we'll call `OpenInApp` with that particular app name.

We need to finalize how the calls from web would look like and for that, we have a meeting scheduled tomorrow

Reply:
We said, that we first will go with an external extension for Web and later try to move it in to the Web repository. 

Currently there are limitations what a Web extension can do [when registering file handlers](https://github.com/owncloud/ocis-wopiserver/blob/main/ui/app.js), therefore I already wrote down requirements which also cover the ones for the AppProvider integration: https://github.com/owncloud/web/issues/5135

So basically we are currently limited by the topics mentioned in this issue.

In my opinion we need some simple map like this in the frondent:

``` json
{
  "application/vnd.oasis.opendocument.text": {
    "appprovider": [
      { "provider": "collabora.example.test", "displayname": "Collabora" },
      { "provider": "onlyoffice.example.test", "displayname": "Only Office" }
    ]
  },
  "application/vnd.oasis.opendocument.spreadsheet": {
    "appprovider": [
      { "provider": "collabora.example.test", "displayname": "Collabora" },
      { "provider": "onlyoffice.example.test", "displayname": "Only Office" }
    ]
  }
}
```

But we should consider to add more information to it, since some appproviders might support opening files from x while other do not. Shares, Federated Shares, ... come into my mind when thinking about that.
Also some appproviders might support multiple modes to open a file: open in read-write-mode and open in read-only-mode.

Reply:
So with the new workflow, we won’t need to list these extensions statically. There’s an endpoint called GetStorageProvides which can be exposed through a REST layer which will be called for each file and the list of supported apps should be populated from that response. And the response is similar to what you mentioned but instead of returning providers for all mime types, we return only those for the passed file.

Not opening shares is very reva specific so I don’t think it’ll fit in CS3APIs. That logic can be implemented in the app drivers and we can return the appropriate status code.

But the option to restrict view modes in apps itself makes a lot of sense. I’ll add it to the PR